src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -687,7 +676,7 @@ void BaseCouplingScheme::advanceTXTWriters()
 
 bool BaseCouplingScheme::reachedEndOfTimeWindow()
 {
-  return math::equals(getThisTimeWindowRemainder(), 0.0, _eps);
+  return math::equals(getNextTimestepMaxLength(), 0.0, _eps) || not hasTimeWindowSize();
```

Comment:
I think this is the most critical change with respect to functionality of `getNextTimestepMaxLength` vs. `getThisTimeWindowRemainder`.